### ISSUE TITLE
Incorrect markdown (tables, spoilers, small text)

### ISSUE THREAD

**Issue by jordan1776** [liftoff-app/liftoff#84]
### Device info

- OS: Android 
- OS version: 11
- Device: Xperia 1 II
- Liftoff version: 0.9.12

### Describe the bug
Things which so not render correctly:

- subscripts
- spoiler text
- tables inside spoiler tags
- bold text

### Steps to reproduce

1. View a post/ comment which contains these tags

### Expected behavior

All markdown should be rendered as intended.

### Screenshots/Screencasts

Lemmy mobile:
![Screenshot_20230625-132550](https://github.com/liftoff-app/liftoff/assets/6100101/f2d7026c-b30f-4d58-a407-5664e4542ff1)

![Screenshot_20230625-142806](https://github.com/liftoff-app/liftoff/assets/6100101/e805edc7-fa24-4e65-b14c-5806b1801314)

Liftoff:

![Screenshot_20230625-132558](https://github.com/liftoff-app/liftoff/assets/6100101/136fe404-44de-4f8d-9c7f-7c8a36c06e27)

![Screenshot_20230625-142754](https://github.com/liftoff-app/liftoff/assets/6100101/34077a10-9add-46bb-97b9-60c25f951c25)

### Additional context

https://lemmy.ml/post/1202746

https://sh.itjust.works/post/312073

**Comment by jordan1776** [comment]
Also there's a lot of slowdown on the spoiler table post but only when scrolling over the table

**Comment by zachatrocity** [comment]
I did some investigation on this one. Its actually an error with the spoiler tag. Table works fine. 

In this post: https://sh.itjust.works/post/312073 there is actually an error with a missing `:::` closing tag for the spoiler, it only works because its at the end of the post. 

Lemmy uses the following tag for spoilers:

```
::: spoiler my spoiler title
my spoiler content
:::
```

But that isnt valid markdown and has to be converted to github web flavored mark down which uses the `<details>` tag. Example:

<details><summary>Spoiler here!</summary>
this is a spoiler!
</details>

And the code to generate it:

```
<details><summary>Spoiler here!</summary>
this is a spoiler!
</details>
```

You can see how they did this in lemmy-ui here: https://github.com/LemmyNet/lemmy-ui/blob/773176084cd380541bbe041de39a4a572554d1a3/src/shared/markdown.ts#L48-L64

I was able to get dart's markdown to parse the lemmy tag and convert it using these inline syntaxes for MarkdownBody:

```
class SpoilerStartSyntax extends md.InlineSyntax {
  //block regex
  // r':::[\s]?spoiler\s+(.*)([\s\S]*):::$'
  SpoilerStartSyntax() : super(r':::[\s]?spoiler\s+(.*)$');

  @override
  bool onMatch(md.InlineParser parser, Match match) {
    parser.addNode(md.Text('<details><summary>'));

    if (match.group(1) != null) {
      // Just use the original matched text.
      parser.addNode(md.Text(match.group(1)!));
    }

    parser.addNode(md.Text('</summary>\n'));
    // parser.
    return true;
  }
}

class SpoilerEndSyntax extends md.TextSyntax {
  SpoilerEndSyntax() : super(':::', sub: '\n </details> \n');
}
```

Then used in MarkdownBody like so:

` inlineSyntaxes: [SpoilerStartSyntax(), SpoilerEndSyntax()],`

I believe this needs to use BlockSyntax instead of inline but it was giving me trouble. I need to focus on other higher priority bugs right now but wanted to leave this info incase some one wants to tackle this ticket.

**Comment by jordan1776** [comment]
Also it seems that spoilers around images don't work. And this one does have a ::: at the end of the comment.

![Screenshot_20230701-134204](https://github.com/liftoff-app/liftoff/assets/6100101/c9cb8f8a-7724-4f21-a499-d674ecfde255)

**Comment by zachatrocity** [comment]
Yeah this entire issue is due to spoiler tag. Tables actually work fine. I've almost got a fix for this

**Comment by jordan1776** [comment]
Nice, thanks for the great dev work!
